integration-tests/src/tests/client/process_blocks.rs

```diff
@@ -1256,58 +1256,40 @@ fn test_bad_orphan() {
 
 #[test]
 fn test_bad_chunk_mask() {
-    // TODO(#10506): Fix test to handle stateless validation
-    if checked_feature!("stable", StatelessValidationV0, PROTOCOL_VERSION) {
-        return;
-    }
-
     init_test_logger();
-    let chain_genesis = ChainGenesis::test();
-    let validators = vec!["test0".parse().unwrap(), "test1".parse().unwrap()];
-    let mut clients: Vec<Client> = validators
-        .iter()
-        .map(|account_id| {
-            let vs = ValidatorSchedule::new()
-                .num_shards(2)
-                .block_producers_per_epoch(vec![validators.clone()]);
-            setup_client_with_synchronous_shards_manager(
-                create_test_store(),
-                vs,
-                Some(account_id.clone()),
-                false,
-                Arc::new(MockPeerManagerAdapter::default()).into(),
-                MockClientAdapterForShardsManager::default().into_sender(),
-                chain_genesis.clone(),
-                TEST_SEED,
-                false,
-                true,
-            )
-        })
-        .collect();
-    for height in 1..5 {
-        let block_producer = (height % 2) as usize;
-        let chunk_producer = ((height + 1) % 2) as usize;
 
-        let ProduceChunkResult {
-            chunk: encoded_chunk,
-            encoded_chunk_parts_paths: merkle_paths,
-            receipts,
-            ..
-        } = create_chunk_on_height(&mut clients[chunk_producer], height);
-        for client in clients.iter_mut() {
-            client
-                .persist_and_distribute_encoded_chunk(
-                    encoded_chunk.clone(),
-                    merkle_paths.clone(),
-                    receipts.clone(),
-                    client.validator_signer.as_ref().unwrap().validator_id().clone(),
-                )
-                .unwrap();
+    // Create a TestEnv with two shards and two validators who track both shards
+    let accounts = vec!["test0".parse().unwrap(), "test1".parse().unwrap()];
+    let num_validators: u64 = accounts.len().try_into().unwrap();
+    let genesis = Genesis::test_sharded(accounts.clone(), num_validators, vec![num_validators; 2]);
+    let mut env = TestEnv::builder(ChainGenesis::new(&genesis))
+        .clients(accounts)
+        .real_epoch_managers(&genesis.config)
+        .nightshade_runtimes(&genesis)
+        .track_all_shards()
+        .build();
+
+    // Generate 4 blocks
+    for height in 1..5 {
+        // The test never goes past the first epoch, so EpochId(11111...) can be used for all calculations
+        let first_epoch_id = &EpochId::default();
```

Comment:
nit: can set outside loop

Reply:
Ok, moved before the loop. I originally put it there because I thought that it would be more readable with the variable right next to the lines that use it. But it can be before the loop as well, I'm fine with both.